### PR TITLE
#187 added search and sorting

### PR DESCRIPTION
**Pull request by RamazanYapparov**
Backend part for #187

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java [hunk 1]
```diff
@@ -80,26 +82,55 @@
...
+        .filter(topic -> !topic.isInternal()
+            || showInternal
+            .map(i -> topic.isInternal() == i)
+            .orElse(true))
+        .filter(topic ->
+            search
+                .map(s -> StringUtils.containsIgnoreCase(topic.getName(), s))
+                .orElse(true))
+        .map(clusterMapper::toTopic)
+        .collect(Collectors.toList());
```
Comment (IldarAlmakaev) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java:105]: Wouldn't be better if we filter topics at first and only then sort them?

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 2]
```diff
@@ -140,6 +140,21 @@ paths:
           required: false
           schema:
             type: integer
+        - name: showInternal
+          in: query
+          required: false
```
Comment (IldarAlmakaev) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:145]: Shouldn't we set default values for these optional parameters?

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 3]
```diff
@@ -1245,6 +1260,13 @@ components:
           items:
             $ref: '#/components/schemas/Topic'
 
+    TopicColumnsToSort:
+      type: string
+      enum:
+        - NAME
+        - OUT_OF_SYNC_REPLICAS
+        - TOTAL_PARTITIONS
+
```
Comment (IldarAlmakaev) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:1269]: ...and the default value here as well?

**Review by IldarAlmakaev — CHANGES_REQUESTED** [review]
Overall, looks good to me.
I left some minor requests and questions. Please take a look at them

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java (hunk 1 above)
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ClusterService.java:105]: Good point, thanks
Fixed

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 2 above)
Comment (RamazanYapparov) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:145]: I intentionally left them as optionals with no defaults
No value is the default value

Moreover using `default` doesn't change the generated code (except for annotations) both for be and fe

Comment (IldarAlmakaev) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:145]: Ok
Let's keep it as is

**Review by IldarAlmakaev — APPROVED** [review]

**Pull request merged**